### PR TITLE
Fix upload rules resource eval logic

### PR DESCRIPTION
**Pull request by tonyjhuang**
### Description

Sets `request.resource` properly when evaluating object CREATE events. 

This was broken in #4250, see `files.ts` in this PR. Previously, we were setting the incoming file metadata as the pre-existing `resource` variable as opposed to the incoming `request.resource` variable before handing off to rules.

 The rest of this PR is adding test coverage for the gap and cleaning up the integration test security rules.

### CODE REVIEW

File: scripts/storage-emulator-integration/tests.ts [hunk 1]
```diff
@@ -129,6 +129,35 @@ async function resetStorageEmulator(emulatorHost: string) {
...
+        if (err instanceof Error) {
+          throw err.message;
+        }
+        throw err;
+      }
+    },
+    filename,
+    text,
+    format ?? "raw",
+    JSON.stringify(metadata ?? {})
```
Comment (tohhsinpei) [inline scripts/storage-emulator-integration/tests.ts:157]: Is there any reason why you stringify and then re-parse?

File: scripts/storage-emulator-integration/tests.ts [hunk 2]
```diff
@@ -787,6 +697,144 @@ describe("Storage emulator", () => {
...
+        it("should upload a file", async function (this) {
+          this.timeout(TEST_SETUP_TIMEOUT);
+
+          const uploadState = await page.evaluate((IMAGE_FILE_BASE64) => {
+            const auth = (window as any).auth as firebase.auth.Auth;
+
+            return auth
+              .signInAnonymously()
+              .then(() => {
+                return firebase
```
Comment (tohhsinpei) [inline scripts/storage-emulator-integration/tests.ts:710]: Can you not use `uploadText` here? Similarly below

**Review by tohhsinpei — COMMENTED** [review]
Looks fine, left a couple small questions

**Review by abhis3 — APPROVED** [review]
lgtm

File: scripts/storage-emulator-integration/tests.ts [hunk 3]
```diff
@@ -697,28 +697,16 @@ describe("Storage emulator", () => {
         );
       });
 
-      describe("#put()", () => {
+      describe.only("#put()", () => {
```
Comment (tohhsinpei) [inline scripts/storage-emulator-integration/tests.ts:700]: Remove

**Review by tohhsinpei — APPROVED** [review]

File: scripts/storage-emulator-integration/tests.ts (hunk 3 above)
Comment (tonyjhuang) [reply scripts/storage-emulator-integration/tests.ts]: done

**Pull request merged**